File: frontend/src/app/services/cloth.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Cloth } from '../model/cloth.model';
import { API_URL } from '../app.constants';

@Injectable({
  providedIn: 'root'
})
export class ClothService {

  constructor(
    private http:HttpClient
  ) { }

  retrieveCloths(clothTypeId, dateFrom, dateTo,colorId, page,size,sort) {
    if (dateFrom != "" && dateTo != "") {
      return this.http.get<Cloth[]>(`${API_URL}/cloths?clothTypeId=${clothTypeId}&dateFrom=${dateFrom}&dateTo=${dateTo}&colorId=${colorId}&page=${page}&size=${size}&sort=${sort}`);
    }else{
      return this.http.get<Cloth[]>(`${API_URL}/cloths?clothTypeId=${clothTypeId}&colorId=${colorId}&page=${page}&size=${size}&sort=${sort}`);
    }
  }

  deleteCloth(clothId){
    return this.http.delete(`${API_URL}/cloths/${clothId}`);
  }

  retrieveCloth(clothId){
    return this.http.get<Cloth>(`${API_URL}/cloths/${clothId}`);
  }

  updateCloth(clothId, cloth){
    return this.http.put(
          `${API_URL}/cloths/${clothId}`
                , cloth);
  }

  createCloth(cloth){
    return this.http.post(
              `${API_URL}/cloths`
                , cloth);
  }
}

File: backend/src/main/java/com/dream/services/ClothService.java
package com.dream.services;

import org.springframework.data.domain.Pageable;

import java.text.ParseException;

import org.springframework.data.domain.Page;

import com.dream.models.Cloth;

public interface ClothService {
	
	Page<Cloth> getCloths(long clothTypeId,String dateFrom, String dateTo,long colorId, Pageable pageable) throws ParseException;
	
	Cloth getClothById(long id);
	
	Cloth createCloth(Cloth Cloth);
	
	Cloth updateCloth(long id, Cloth Cloth);
	
	void deleteCloth(long id);
}
